### ISSUE TITLE
You might find this interesting in regard to Softmax

### ISSUE THREAD

**Issue by jukofyork** [TheoLeeCJ/SemIf#1]
> Softmax over allowed tokens is conditional on the supplied alternatives; it is not calibrated operational confidence.

I haven't time to read all this but you might find this interesting:

https://en.wikipedia.org/wiki/Luce%27s_choice_axiom (and the related https://en.wikipedia.org/wiki/Independence_of_irrelevant_alternatives).

and:

https://en.wikipedia.org/wiki/Platt_scaling

This also works for multinominal logit (aka Softmax) and is essentially just the same as altering the temperature (or equivalently multiplying `lm_head` by `1/temperature`). Note that the offset `B` for the binary case can't be applied to Softmax, as its output is invariant to shifting.

You might also find different tasks require different calibration temperature, but there should be an optmimal temperature that produces well calibrated outputs for a given task and you can probably even find it online by minimisation of error deltas via a simple controller or RL, etc.
